probe-rs/src/debug/mod.rs

```diff
@@ -243,8 +288,16 @@ pub struct DebugInfo {
 }
 
 impl DebugInfo {
-    pub fn from_raw(data: &[u8]) -> Self {
-        let object = object::File::parse(data).unwrap();
+    /// Read debug info directly from a ELF file.
+    pub fn from_file<P: AsRef<Path>>(path: P) -> Result<DebugInfo, Error> {
+        let data = std::fs::read(path)?;
+
+        DebugInfo::from_raw(&data)
+    }
+
+    /// Parse debug information directly from a buffer containing an ELF file.
+    pub fn from_raw(data: &[u8]) -> Result<Self, Error> {
+        let object = object::File::parse(data).map_err(|e| Error::DebugData(e))?;
```

Comment:
Is this `map_err` really required? Isn't from automatically inferred with the thiserror derive?

Reply:
Yes, it's necessary. The actual Error is of type `'static str`, but I didn't want to add generic `From` implementation for all `'static str`.

Reply:
Looks like it is required in this case because the `object` crate does not use proper error types. `object::File::parse` returns `Result<_, &'static str>` and there is no Error impl for &amp;'static str.

https://docs.rs/object/0.17.0/object/read/struct.File.html#method.parse

Reply:
They have an issue to track improving the error handling in `object`: https://github.com/gimli-rs/object/issues/194; I left a comment.